File: test/fixtures/rules/typescript/ternary-same-value.ts
export function positives(n: number, fallback: string): string[] {
	return [
		n > 0 ? fallback : fallback, // SLOP
	];
}

// Must NOT fire: the branches genuinely differ.
export function negatives(n: number, a: string, b: string): string[] {
	return [n > 0 ? a : b, n > 0 ? a : "x"];
}

File: test/fixtures/rules/swift/ternary-same-value.swift
func positives(_ n: Int, _ fallback: String) -> [String] {
	return [
		n > 0 ? fallback : fallback, // SLOP
	]
}

// Must NOT fire: the branches genuinely differ.
func negatives(_ n: Int, _ a: String, _ b: String) -> [String] {
	return [n > 0 ? a : b, n > 0 ? a : "x"]
}
